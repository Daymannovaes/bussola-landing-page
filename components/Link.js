import Link from 'next/link';
import css from 'styled-jsx/css';

import { colors } from '../config/style-guide';

export default ({ children, noprefetch, nostyle, ...rest }) => (
    <React.Fragment>
        <style jsx>{`
            a.styled-link {
                color: ${colors.lightblue};
                text-transform: uppercase;
                transition: 200ms all;
            }

            a.styled-link:hover {
                background: #e9e8e8;
            }
        `}</style>

        <Link prefetch={!noprefetch}  {...rest}>
            <a className={nostyle ? '' : 'styled-link'}>
                {children}
            </a>
        </Link>
    </React.Fragment>
);
