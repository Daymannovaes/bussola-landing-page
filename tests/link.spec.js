import React from 'react';
import Link from '../components/Link';
import renderer from 'react-test-renderer';

import Router from 'next/router';
const mockedRouter = { push: () => {}, prefetch: () => {} };
Router.router = mockedRouter;

describe('link.spec.js', () => {
    it('snapchot', () => {
        const component = renderer.create(
            <Link href="/about">About</Link>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        console.log(tree);
        expect(tree.props.className).toContain('styled-link');
    });
});
