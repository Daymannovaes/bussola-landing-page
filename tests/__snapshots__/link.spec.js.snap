// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link.spec.js snapchot 1`] = `
<a
  className="jsx-1049726718 styled-link"
  href="/about"
  onClick={[Function]}
>
  About
</a>
`;
